fiftyone/plugins/secrets.py

```diff
@@ -118,6 +125,12 @@ def __init__(self, secrets_dict, operator_uri=None, resolver_fn=None):
     def __eq__(self, other):
         return self.__secrets == other
 
+    def __setattr__(self, key, value):
+        if key == "__secrets" or key == "__required_keys":
```

Comment:
This isn't necessary. Double underscore attributes already aren't accessible via `d.__secrets` from outside the class. Python does weird name mangling that makes them available as `d._SecretsDictionary__secrets`.

Either way, don't waste effort trying to prevent things from happening if users modify private variables. It's Python. If the user really wants to modify private variables then they can mess things up infinitely worse than just overriding secrets.

Reply:
I don't see it as wasted effort because at the end of the day if they mess it up, it comes back to me to debug. I'd rather prevent mutation from the start rather than allow it.